### ISSUE TITLE
ViewManagerResolver returns null for RNUnity or RCTRNUnityView

### ISSUE THREAD

**Issue by Omaryehya21** [azesmway/react-native-unity#142]
Hello,

I'm trying to integrate a Unity project into my Expo app using the @azesmway/react-native-unity plugin. I followed the steps in the documentation and also referred to this [article](https://ibjects.medium.com/react-native-unity-3d-tutorial-6ef230df81a8).

Steps Taken:

1. Installed the package version @1.0.7.
2. Created a basic stack with the following code in Unity.tsx in the tabs folder:
```
import React, { useRef } from 'react';
import UnityView from '@azesmway/react-native-unity';
import { View } from 'react-native';

const Unity = () => {
  const unityRef = useRef<UnityView>(null);

  return (
    <View style={{ flex: 1 }}>
      <UnityView
        ref={unityRef}
        style={{ flex: 1 }}
      />
    </View>
  );
};
export default Unity;
```
3. Ran the app using Expo and tried to open the UnityView. 

Error:
When attempting to render UnityView, I consistently get the following error:
`ViewManagerResolver returns null for either RNUnity or RCTRNUnityView`

What seems the problem I checked the dependency and the library is there.

**Comment by TruAmbition** [comment]
If you are using Xcode 16, please upgrade the package version 1.0.11. Please note, we don't officially support Expo. This package does support react-native projects without Export framework .

**Comment by Omaryehya21** [comment]
Sorry for the late reply. I am currently using VSCode on Windows, and I even attempted to use Expo pre-build to access the Android Gradle and followed the steps in the README file, but the same issue persists. I initially assumed that Expo was supported, as the README mentions after the third step for running on Android: "Remove ` <intent-filter>...</intent-filter>`  from `<YOUR_RN_PROJECT>/unity/builds/android/unityLibrary/src/main/AndroidManifest.xml` at unityLibrary to leave only the integrated version." It further states, "If you're using Expo, you're done. The built-in Expo plugin will handle the rest. If you're not using Expo, you'll need to follow the steps below."

This left me a bit confused, as I am using the latest React Native/Expo package. Any clarification would be appreciated.

**Comment by EvvTim** [comment]
Hello,

I am experiencing the same issue when trying to use Expo with the @azesmway/react-native-unity plugin. Specifically, I get the error:
ViewManagerResolver returns null for either RNUnity or RCTRNUnityView.

From my understanding, the documentation mentions that “if you’re using Expo, you’re done,” which implies that Expo should be supported. However, based on the clarification in this thread, Expo is not officially supported anymore.

I believe it might be helpful to update the documentation to explicitly state that Expo is not currently supported, or at least remove the references suggesting compatibility with Expo. This could save developers a lot of time and effort by avoiding confusion.

Thanks!

**Comment by Omaryehya21** [comment]
Hello @EvvTim,
Have you resolved this issue? We’ve already built the front end using the latest Expo SDK (v52). I also tried using a pre-built Expo project and running it directly on Android (instead of through Expo Go), but the issue persists.

Have you discovered any workarounds for this?

**Comment by EvvTim** [comment]
> Hello @EvvTim, Have you resolved this issue? We’ve already built the front end using the latest Expo SDK (v52). I also tried using a pre-built Expo project and running it directly on Android (instead of through Expo Go), but the issue persists.
> 
> Have you discovered any workarounds for this?

You must run `npx expo prebuild `to generate native code for your project and then follow the step-by-step instructions from the README.

This worked for me.

**Comment by Omaryehya21** [comment]
Hello @EvvTim , I tried using npx expo prebuild and followed the step-by-step instructions, but the same errors still occur. I used `npx expo start` to run the project—maybe that's the problem? Or could there be another issue?

**Comment by Omaryehya21** [comment]
using npx expo prebuild to genrate native code worked.

**Comment by TackJordy** [comment]
@Omaryehya21 Did you use npx expo prebuild and then followed the steps from the readme, or something else too? I've tried all the possibilities that are listed here but can't get it working in expo. I keep getting a gray screen.

**Comment by Omaryehya21** [comment]
@TackJordy Yes, I followed the steps in the readme after using the npx expo prebuild command. If your on android, I faced the same issue where the unity did not launch, The issue was that there was a difference between the unity ndk and android ndk. What I did was overwrite the android ndk version in the file 'unity\builds\android\unityLibrary\build.gradle' to match the react native version as such
`android {
    namespace "com.unity3d.player"
    ndkPath "path/to/ndk/version" 
    ndkVersion "26.1.10909125" <--- match ndk version found in {root}/android/build.gradle.
    compileSdk 35
    buildToolsVersion = "34.0.0"
`

**Comment by TackJordy** [comment]
Thanks, might get my hands on an Android then, I'm trying to make it work on ios, but keep struggling, even with a completely new expo project. I'm doing the prebuild and following the steps to the point

**Comment by Omaryehya21** [comment]
@TackJordy Are you running the app using Expo go? if so I don't think it will work, the workaround requires to build it as a react native cli project. If you're using ios you need xcode to build and run the app, if not and still have the same issue, try configure it for android and use an AVD (Android virtual device) with a simple unity project (render a cube), build and run the app from android studio.
